docs: addressing common vulnerabilities (#608)
ansys/pyansys-dev-guide: doc/source/how-to/vulnerabilities.rst

 > [!CAUTION]
 > Do not use GitHub issues to report any security vulnerabilities.

> 

Reviewer: ```suggestion
```

 - Any specific configuration settings needed to reproduce the problem
 - Step-by-step guidance to reproduce the problem
 - The exact location of the problematic source code, including tag, branch, commit, or a direct URL

>  - The potential consequences of the vulnerability, along with a description of how an attacker could take
>    advantage of the issue

Reviewer: ```suggestion
 - The potential consequences of the vulnerability, along with a description of how an attacker could take advantage of the issue
```
Ref is single line

Additionally, the security advisory should be published on the repository's **Security** tab. This
advisory should include the same information as the changelog note, in other words the CVE number,
the date of the advisory, and the status of the advisory (such as published or withdrawn).

A reference of a published security advisory can be found here: `PyAnsys Geometry subprocess
advisory`_. This advisory was published in the PyAnsys Geometry repository and includes
information about a vulnerability in which users could execute arbitrary code on the system by
using one of this library's functions.

Ensuring compliance across the PyAnsys ecosystem
------------------------------------------------

The PyAnsys Core team is responsible for ensuring that the ``ansys/actions/check-vulnerabilities``
action is up to date and that it is being used in all PyAnsys repositories considered as libraries
(that is, Python packages shipped to PyPI). Repository maintainers are responsible for ensuring
that the action is implemented correctly and that the results are reviewed regularly.

Addressing common vulnerabilities
---------------------------------

> When developing Python applications, it's essential to be aware of common vulnerabilities that can

Reviewer: ```suggestion
When developing Python applications, it is essential to be aware of common vulnerabilities that can
```
Avoid contractions

not be used in Python code. Using these functions can lead to security vulnerabilities and
should be avoided. Refer to the `blacklists Bandit documentation`_ for detailed information on
`Bandit` tool outputs.

**Bandit blacklist**

The `Bandit` tool provides a blacklist of known vulnerable functions and methods that should
not be used in Python code. Using these functions can lead to security vulnerabilities and
should be avoided.

Address each requested changes proposed by `Bandit` to ensure that your code is secure.
You can find information on how to improve your code in the `blacklists Bandit documentation`_.

**subprocess command injection**

The `subprocess` module can be vulnerable to command injection if user input is not properly
sanitized. This can lead to arbitrary command execution, which is a significant security risk.

To mitigate this risk, you should:

- avoid using the `subprocess` module to execute shell commands with user input, as it can lead
  to command injection vulnerabilities.
- if the previous point is not possible, you need to disable the `shell=True` argument in 
  `subprocess.run()` or similar functions, as it allows for shell injection attacks.

> By removing the `shell=True` argument, a list will be needed to pass the command and its

Reviewer: ```suggestion
By removing the `shell=True` argument, a list is needed to pass the command and its
```
